internal/workerpool/pool_test.go

```diff
@@ -83,4 +95,41 @@ func TestRun(t *testing.T) {
 			time.Sleep(time.Millisecond)
 		}
 	})
+
+	t.Run("returns error if a task returns error", func(t *testing.T) {
+		ng0 := runtime.NumGoroutine()
+		ctx := context.Background()
+		wantErr := fmt.Errorf("error-%d", rand.Int())
+		nextTask := func(context.Context) (workerpool.Task, error) {
+			task := func(context.Context) error {
+				return wantErr
+			}
+			return task, nil
+		}
+		err := workerpool.Run(ctx, 1, nextTask)
+		if err != wantErr {
+			t.Errorf(`workerpool.Run returned "%v" for error, want "%v"`, err, wantErr)
+		}
+		t.Log("waiting for goroutines to return")
+		for runtime.NumGoroutine() != ng0 {
```

Comment:
This might turn out to be fragile. Who knows when the runtime decides to create a long-lived goroutine?

But leave it and I guess we'll see.

Reply:
That's why I don't call `t.Parallel()` at the top of these tests. I've never had a problem with this, but I guess if I do then I'll learn something.